src/module_wrap.cc

```diff
@@ -980,40 +985,50 @@ Maybe<URL> ResolveExportsTarget(Environment* env,
       Local<Array> target_arr = target.As<Array>();
       const uint32_t length = target_arr->Length();
       if (length == 0) {
-        if (throw_invalid) {
-          ThrowExportsInvalid(env, pkg_subpath, target, pjson_url, base);
-        }
+        ThrowExportsInvalid(env, pkg_subpath, target, pjson_url, base);
         return Nothing<URL>();
       }
       for (uint32_t i = 0; i < length; i++) {
         auto target_item = target_arr->Get(context, i).ToLocalChecked();
-        if (!target_item->IsArray()) {
+        {
+          TryCatchScope try_catch(env);
           Maybe<URL> resolved = ResolveExportsTarget(env, pjson_url,
-              target_item, subpath, pkg_subpath, base, false);
-          if (resolved.IsNothing()) continue;
+              target_item, subpath, pkg_subpath, base);
+          if (resolved.IsNothing()) {
+            CHECK(try_catch.HasCaught() && !try_catch.Exception().IsEmpty());
```

Comment:
I don’t think this is necessarily true, though – execution could have terminated before this point. Can you handle  `!try_catch.HasCaught()` (by just returning `Nothing<URL>()` directly in that case)?

Reply:
The contract for the `ResolveExportsTarget` function (and many other functions in this file) is that the `Nothing` return path will always correspond to when an error was thrown. Anything else should be considered an internal logic / assertion failure.

Reply:
Yeah, sorry, I think you’re right in that `try_catch.HasCaught()` is guaranteed to be true at this point; but if `try_catch.HasTerminated()` is true, you can’t perform anything meaningful with `try_catch.Exception()` (I *think* it returns an internal special V8 object in that case, but I’m not fully sure about that).